File: backend/account-manager/src/main/java/com/jay/accountmanager/model/Income.java
package com.jay.accountmanager.model;

import java.math.BigDecimal;
import java.time.LocalDate;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "incomes")
public class Income {
    
	@Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private BigDecimal amount;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private IncomeType type;

    @Enumerated(EnumType.STRING)
    private RecurrenceType recurrenceType;

    private Integer recurrenceDay; // Day of month for monthly, or day of week for weekly
    
    private LocalDate nextDueDate;
    
    @Column(nullable = false)
    private String currency;

    @ManyToOne
    @JoinColumn(name = "person_id", nullable = false)
    private Person person;

    private String description;
    
}

File: frontend/src/types/income.ts
export enum IncomeType {
    SALARY = 'SALARY',
    RENT = 'RENT',
    FREELANCE = 'FREELANCE',
    INVESTMENT = 'INVESTMENT',
    OTHER = 'OTHER'
}

export enum RecurrenceType {
    ONCE = 'ONCE',
    WEEKLY = 'WEEKLY',
    BIWEEKLY = 'BIWEEKLY',
    MONTHLY = 'MONTHLY',
    CUSTOM = 'CUSTOM'
}

export interface Income {
    id: number;
    name: string;
    amount: number;
    type: IncomeType;
    recurrenceType: RecurrenceType;
    recurrenceDay?: number;
    nextDueDate?: string;
    currency: string;
    description?: string;
    isActive: boolean;
    createdAt: string;
    updatedAt: string;
}

export interface IncomeRequest {
    name: string;
    amount: number;
    type: IncomeType;
    recurrenceType: RecurrenceType;
    recurrenceDay?: number;
    nextDueDate?: string;
    currency: string;
    description?: string;
    accountId?: number;
}
